### DISCUSSION TITLE
Support for Shelly PM Mini gen3

### DISCUSSION THREAD

**Discussion by Bodengriller** [arendst/Tasmota#21909] [Support request] [answered]
Is Shelly PM Mini gen3 supported?
[https://www.shelly.com/de/products/shop/shelly-pm-mini-gen3](https://www.shelly.com/de/products/shop/shelly-pm-mini-gen3)
Cant find any Info about this

Could configuration for Shelly Plus PM Mini be used?
[https://templates.blakadder.com/shelly_plus_pm_mini.html](https://templates.blakadder.com/shelly_plus_pm_mini.html)

**Comment by sfromis** [comment]
They do seem to be *very* similar, except that the gen3 has 8 MB of flash. I'd indeed suggest trying if that template works for you. If not, no harm done.

**Reply by Bodengriller** [reply]
I thought the same thing.
So I'll just order one and give it a try. 
Feedback will follow

**Comment by arendst** [comment]
This is the valid template ```{"NAME":"Shelly Mini1PMG3","GPIO":[576,32,0,4736,0,224,3200,8161,0,0,192,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}``` and ``AdcParam1 2,10000,10000,3350``

**Reply by sfromis** [reply]
Also, https://github.com/tasmota/mgos32-to-tasmota32 includes gen3 devices "Mini1PMG3" and "MiniPMG3". Beware of the warning about trouble with new Shelly fw 1.4.x devices.

**Reply by Bodengriller** [reply]
Thank you :)

It would be nice if the devices could be included in the compatibility list with these notes 
@arendst  where did you find that tamplate?

@sfromis 
just for my understanding
This problem only occurs when trying to update via the OTA, right?
Tasmota should be able to be flashed as usual via the serial connection without any problems?

**Reply by sfromis** [reply]
Right.

**Reply by Bodengriller** [reply]
Now I'm only wondering about two things...

1. is it even necessary to use [mgos32-to-tasmota32-MiniPMG3.zip](https://github.com/tasmota/mgos32-to-tasmota32/releases/download/v13.4.1/mgos32-to-tasmota32-MiniPMG3.zip) if you flash via serial anyway?
I always thought mgos32-to-tasmota was just the convenient OTA way to leave the serial adapter in the drawer
Is the bootloader also replaced when using the [Tasmota-Web-Flasher](https://tasmota.github.io/install/) or is the workaround via mgos32-to-tasmota necessary?

2. Where did @arendst find this template? :D
Even with the template I can't find it anywhere - is there still a secret database?

**Reply by arendst** [reply]
1. It's either serial OR mgos to Tasmota.
2. After elaborate investigation of the hardware I came to this template (as I came to mosts templates for Tasmota).

**Reply by Bodengriller** [reply]
Ah okay, thank you.
But you can't find this template published yet, can you?
Many users have drifted off into the Shelly-Universe, but I think there are still some who love Tasmota and want to use it, just like me.
The template should also be interesting for them

BTW: I think there's a little mistake: "Shelly Mini_1_PMG3" should be "Shelly MiniPMG3" (without the 1), shouldn't it?

**Reply by arendst** [reply]
On the device it clearly says "Shelly 1PM Mini Gen3" where the "1" points to the many phases it supports.

"SomeOne" must be willing to add the template to https://templates.blakadder.com/.

**Reply by arendst** [reply, marked as answer]
Ah I see what you mean. You're talking about the "Shelly PM Mini Gen3". In that case you need a different template. Let me try to find it from the pile of rubbish I have lying around...

Found it. It's in the BL09xx driver:
```
{"NAME":"Shelly PlusPMMini","GPIO":[576,32,0,4736,0,0,3200,8161,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1,"CMND":"AdcParam1 2,5600,4700,3350}
```

**Reply by Bodengriller** [reply]
hehe OK, that explains the misunderstanding.
Yes, I only meant the PowerMeter, without the relay.

So the same template from the older Version "Shelly Plus PM Mini" works - as suspected.
Thank you very much.

I will add this on Blackadder as soon as my devices arrive tomorrow and I have successfully flashed and tested them.

**Reply by FransOv** [reply]
> On the device it clearly says "Shelly 1PM Mini Gen3" where the "1" points to the many phases it supports.
> 
> "SomeOne" must be willing to add the template to https://templates.blakadder.com/.

I submitted a template yesterday for the 1PM Mini Gen 3. The template can be used for the PM only and the relay without PM to, just drop the switch and relay or the PM settings. The Mini Gen 3 can be opened easily. The pads for serial flashing are awfully small and close together. Luckily the ones I bought were version 1.3.3 so could be flashed OTA.

**Reply by TaRga51170** [reply]
> This is the valid template `{"NAME":"Shelly Mini1PMG3","GPIO":[576,32,0,4736,0,224,3200,8161,0,0,192,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}` and `AdcParam1 2,10000,10000,3350`
Would anyone have the template for Shelly1 GEN3 please.  I modified an existing one but not sure if this is correct.
{"NAME":"Shelly1 Gen3","GPIO":[320,0,0,0,192,224,1,1,0,0,193,0,0,0,0,1,0,0,0,0,0,32,224,0,0,0,0,0,4736,4705,0,0,0,0,0,0],"FLAG":0,"BASE":18}

**Comment by Bodengriller** [comment]
I have now successfully flashed 4 Shelly MiniPM G3 with Tasmota.
2x per serial
2x per mgos32-to-tasmota32
Both work.

Now two things confuse me:
1. the memory:
After flashing directly via the webflasher using serial, the flash memory is 4416KB
After flashing via mgos32-to-tasmota32 the memory was somewhere around 1100KB (I forgot to write down the exact value)
I have now set all shellys to 4416KB.
The maximum you can select with the up/down buttons is 5248KB.
But the shellys should have 8MB flash (=8192KB)
I assume that the Tasmota app itself is also in flash - so it has to be added:
5248+2880=8128
Somehow 64KB are still missing to 8192KB
![image](https://github.com/user-attachments/assets/1d3669f8-957f-4e1b-85c4-c489d415f690)

2. the temperature:
All four Shellys show me a temperature >60°C.
That really confuses me.

I used SetOption146 1 to display the CPU temperature - this is approx. 66°C for all 4 Shellys
Also quite high, in my opinion.

The [Shelly PM Mini Gen3](https://devices.esphome.io/devices/Shelly-PM-Mini-Gen3) is already listed in the ESPHome project.
Likewise the predecessor [Shelly Plus PM Mini](https://devices.esphome.io/devices/Shelly-Plus-PM-Mini) and the [Shelly 1PM Mini Gen3 ](https://devices.esphome.io/devices/Shelly-1PM-Mini-Gen3) listed by @arendst  above.
All three models use the same values for temperature calibration there.
@arendst  had two different paramers above for 
Shelly 1PM Mini Gen3: `AdcParam1 2,10000,10000,3350`
and 
Shelly Plus PM Mini: `AdcParam1 2,5600,4700,3350`
I don't quite understand the background to the differences yet.
But when I use `AdcParam1 2,10000,10000,3350` instead of 
simply test `AdcParam1 2,5600,4700,3350`, the displayed temperature rises to >70°C
Maybe the displayed temperature of >60°C is correct?

I assume that the values `AdcParam1 2,5600,4700,3350` are correct, as they seem to have been stored in the blackadder DB for the [Shelly Plus 1PM Mini](https://templates.blakadder.com/shelly_plus_1pm_mini.html) and the [Shelly Plus PM Mini](https://templates.blakadder.com/shelly_plus_pm_mini.html).
But >60°C?! hmm....looks very high to me.

How can I find out without an IR camera?
Of course, I didn't pay attention to this beforehand in the original Shelly interface and flashed tasmota directly onto the boxes, as I should have done :D 
![image](https://github.com/user-attachments/assets/7a068a8f-fae5-4304-bd42-3003562be4e7)

**Reply by Bodengriller** [reply]
BTW in both ways (Serial and mgos32-to-tasmota32) I used the auto-config for "Shelly PlusPMMini" and renamed the Template to
"Shelly MiniPMG3"

So what I would post to blakadder is the following:
`{"NAME":"Shelly MiniPMG3","GPIO":[576,32,0,4736,0,0,3200,8161,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1,"CMND":"AdcParam1 2,5600,4700,3350"}`

**Reply by sfromis** [reply]
A bit of space is being used outside of the partitions, like for settings.

When it comes to temperature value, don't assume that *anything* is correct, until positively verified. That includes the value you get with SetOption146 enabled, this is disabled for a reason, which is that there are too many cases of bad numbers there.

About the temperature as such, while I've not used that specific device, my experience with other Shelly PM devices is that they feel warm to the touch, when on mains power, meaning that the inside board temperature will be a **lot** higher than what I can feel on the outside plastic.

Maybe you need to "calibrate" your expectations for what's "too hot" :fire:

Check the temperature reading right after plugging in, after sufficient time to get fully to room temperature (that's longer than you feeling the plastic to be warmer than ambient).

**Reply by Jason2866** [reply]
When measured temperature directly on the chip is above 95 Celsius it is time to start worrying.

**Reply by Bodengriller** [reply]
The temperature seems to be halfway right.
After the Shellys have now been without power for over 1.5 hours at a room temperature of 26°C, they start with a displayed temperature of approx. 32°C and increase directly to >55°C again in the first 2 minutes.
strange.
I would now be interested to know whether this temperature is displayed at all in the Shelly app or whether it simply serves as overtemperature protection (for whatever reason that would make sense with the MiniPM...with the "normal" Shelly it still makes sense to switch off the relay at some point).

BTW my Shelly S sockets (also Tasmota) show a temperature of approx. 36°C - this also seems plausible for the built-in circuit board measurement

**Reply by sfromis** [reply]
Internal temperature sensing has no use case outside of overtemperature protection. I've not tried viewing things in the Shelly app, but I'd not be surprised if they made a good choice of not confusing the user with such things, as people tend to have beliefs about what is "too much" ;-)

**Comment by arendst** [comment]
Gents, this temp observation is an ever returning issue with always the same answer. The sensor is placed below or next to the esp chip hence the correct high temperature.

It's good for overtemp detection and that's exactly what tasmota does by default; it powers off from 90 degrees centigrade. Again configrable by a SetOption (both powering off AND overtemp value.

Let's call this the last time we discussed shelly internal temperatures.

**Reply by Bodengriller** [reply]
Sorry, I was just wondering why a temperature sensor is installed at all in a pure PowerMeter - it can't switch anything off in the event of overtemperature.
Thanks for the clarification, I'll close this thread :)

**Comment by andrea-adup** [comment]
Does anyone have any idea why Shelly PM Mini gen3 when connected to the air conditioner and this is off shows 0 Watt? I have a Shelly 1pm plus connected to another air conditioner and when the air conditioner is off it shows 0.5-0.9 watts. Can anyone explain? Any help is welcome.

**Reply by sfromis** [reply]
Did you calibrate both devices the same way? 

Even if the answer is "yes", these are not to be regarded as high-precision devices, and self-consumption when the device is running could also factor in.

Holding my hand on an idling Shelly Plus 1PM, I can clearly feel that it is noticeably above room temperature, meaning that it burns some power. I've not got the Shelly PM Mini gen3, but with a different build, it is not a given that it will be as warm, have the same vampire power drain.

Of course, it is not a given that "off" means the same for both AC units, which also may have a vampire power drain, depending on details....

**Comment by andrea-adup** [comment]
sorry I'm a beginner and I don't understand what you mean by calibrate. I just connected the shelly and that's it.

I understand that they can have different tolerances but 2 out of 2 seems strange to me. For the temperature, the shelly mini are much higher.

![Schermata 2025-01-11 alle 22 28 18](https://github.com/user-attachments/assets/192bd156-1034-414c-8336-eaf9d5b4f8c2)

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Power-Monitoring-Calibration/

And different devices may not have similar hardware design about measuring self-consumption. You can argue both ways for whether this "should" be included.

**Reply by andrea-adup** [reply]
thanks, I read the Power monitoring Calibration page

**Comment by SHA-NDY** [comment]
@Bodengriller 
Hi, I tried to flash the mentioned Shelly 1PM Mini Gen3 via serial. Flashing seems to work fine but afterwards I can't see the Tasmota-Hotspot to configure Wifi.
I flashed the factory binary https://ota.tasmota.com/tasmota32/release/tasmota32c3.factory.bin with ESP_flasher and with esptool but nothing happens to Wifi-Hotspot after successful flash.
I flash connected to a USB-UART adapter and afterwards I connect AC power to wait for the Hotspot but as mentioned: no Hotspot there.
Any ideas what I'm missing?
